ZenPacks/zenoss/HttpMonitor/http.py

```diff
@@ -54,6 +55,14 @@ def _getIp(self, response):
         return self.request()
 
     def makeURL(self):
+        if not isip(self._ipAddr):
+            try:
+                self._ipAddr = getHostByName(self._ipAddr)
```

Comment:
You can do this in the `connect` method below.  There's already code to resolve the `hostname` value.  Just add to the chain of `Deferred` objects to resolve the `ipAddr` value as well.  That way all the DNS lookups will asynchronous.